retrofit/src/main/java/retrofit2/RequestAction.java

```diff
@@ -202,13 +203,17 @@
     @Override void perform(RequestBuilder builder, T value) {
       if (value == null) return; // Skip null values.
 
-      RequestBody body;
-      try {
-        body = converter.convert(value);
-      } catch (IOException e) {
-        throw new RuntimeException("Unable to convert " + value + " to RequestBody", e);
+      if (MultipartBody.Part.class.isAssignableFrom(value.getClass())) {
```

Comment:
This is known in the `RequestFactoryParser` and should have an action chosen for the type, not as just-in-time behavior.